src/main/java/seedu/patientist/logic/commands/AddPatientStatusCommand.java

```diff
@@ -0,0 +1,96 @@
+package seedu.patientist.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.patientist.logic.parser.CliSyntax.PREFIX_STATUS;
+
+import java.util.List;
+
+import seedu.patientist.commons.core.Messages;
+import seedu.patientist.commons.core.index.Index;
+import seedu.patientist.logic.commands.exceptions.CommandException;
+import seedu.patientist.model.Model;
+import seedu.patientist.model.Patientist;
+import seedu.patientist.model.person.Person;
+import seedu.patientist.model.person.patient.Patient;
+import seedu.patientist.model.person.patient.PatientStatusDetails;
+import seedu.patientist.model.ward.Ward;
+
+/**
+ * Adds a status to the patient identified using it's displayed index from the patientist book.
+ */
+public class AddPatientStatusCommand extends Command {
+    public static final String COMMAND_WORD = "addpatstatus";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+                                               + ": Adds a status to the patient identified by the index number used in"
+                                               + " the displayed person list.\n"
+                                               + "Parameters: "
+                                               + "INDEX (must be a positive integer) "
+                                               + PREFIX_STATUS + "STATUS\n"
+                                               + "Example: " + COMMAND_WORD + " "
+                                               + "1 "
+                                               + PREFIX_STATUS + "Feeling fine";
+
+    public static final String MESSAGE_ADD_STATUS_SUCCESS = "Added status %1$s to %2$s.";
+    public static final String MESSAGE_NOT_PATIENT = "Person selected is not a Patient.";
+
+    private final Index targetIndex;
+    private final List<PatientStatusDetails> statusDetails;
+
+    /**
+     * Creates an AddPatientStatusCommand to add the specified {@code PatientStatusDeatils} into the target Patient.
+     */
+    public AddPatientStatusCommand(Index targetIndex, List<PatientStatusDetails> statusDetails) {
+        this.targetIndex = targetIndex;
+        this.statusDetails = statusDetails;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToAdd = lastShownList.get(targetIndex.getZeroBased());
+
+        if (!(personToAdd instanceof Patient)) {
+            throw new CommandException(MESSAGE_NOT_PATIENT);
+        }
+
+        Ward ward = null;
+
+        for (String wardName : model.getWardNames()) {
```

Comment:
Instead of doing this would it be easier to invoke the method `model.hasPerson(personToAdd)`? The lines above check if the person is a patient so this logic should still hold if I'm not wrong.

Reply:
I'm doing this method because I need to get the ward that the patient is in. Model does this behind the scenes anyway.

Reply:
Yeah fair enough.